test/user-test.js

```diff
@@ -0,0 +1,121 @@
+import chai from 'chai';
+const expect = chai.expect;
+import users from './sample-test-data/user-test-data';
+import bookings from './sample-test-data/bookings-test-data';
+import rooms from './sample-test-data/rooms-test-data';
+import User from '../src/classes/User'
+
+
+
+describe('User', () => {
+
+  let user;
+  let booking1 = bookings[3];
+  let booking2 = bookings[4];
+  let booking3 = bookings[0];
+  let bookingData = [bookings[0], bookings[1], bookings[2], bookings[4], bookings[5]];
+
+  beforeEach(() => {
+    user = new User(users[0], bookings, rooms);
+  });
+
+  it('should be a function', () => {
+    expect(User).to.be.a('function');
+  });
+
+  it('should be an instance of user', () => {
+    expect(user).to.be.an.instanceof(User);
+  });
+
+  it('should have an id', () => {
+    expect(user.id).to.equal(9);
+  });
+
+  it('should have a name', () => {
+    expect(user.name).to.equal('Faustino Quitzon');
+  });
+
+  it('should default to no bookings', () => {
+    expect(user.bookings.length).to.equal(0);
+    expect(user.bookings).to.deep.equal([]);
+  });
+
+  it('should have bookings', () => {
+    expect(user.bookings.length).to.equal(0);
+
+    user.setBookings(bookingData);
+    expect(user.bookings.length).to.equal(3);
+  });
+
+  it('should be able to add bookings', () => {
+    user.setBookings(bookingData);
+
+    expect(user.bookings.length).to.equal(3);
+
+    user.addBooking(booking1);
+
+    expect(user.addBooking).to.be.a('function');
+    expect(user.bookings.length).to.equal(4);
+    expect(user.bookings[0].id).to.equal('5fwrgu4i7k55hl6sz');
+    expect(user.bookings[0].userID).to.equal(9);
+    expect(user.bookings[0].date).to.equal('2022/04/22');
+    expect(user.bookings[0].roomNumber).to.equal(15);
+    expect(user.bookings[0].roomServiceCharges).to.deep.equal([]);
```

Comment:
This may be too many tests in one it block. You may not need all of these tests here. Or maybe you could break these out into multiple it blocks.

Reply:
I agree. Looks pretty cluttered. I'll break these into separate 'it' blocks. Thank you for pointing this out.